gfx/drivers_context/drm_ctx.c

```diff
@@ -90,7 +90,89 @@ struct drm_fb
    uint32_t fb_id;
 };
 
+// https://www.raspberrypi.org/documentation/configuration/config-txt/video.md
+typedef struct hdmi_timings
+{
+   int h_active_pixels; // horizontal pixels (width)
+   int h_sync_polarity; // invert hsync polarity
+   int h_front_porch;   // horizontal forward padding from DE acitve edge
+   int h_sync_pulse;    // hsync pulse width in pixel clocks
+   int h_back_porch;    // vertical back padding from DE active edge
+   int v_active_lines;  // vertical pixels height (lines)
+   int v_sync_polarity; // invert vsync polarity
+   int v_front_porch;   // vertical forward padding from DE active edge
+   int v_sync_pulse;    // vsync pulse width in pixel clocks
+   int v_back_porch;    // vertical back padding from DE active edge
+   int v_sync_offset_a; // leave at zero
+   int v_sync_offset_b; // leave at zero
+   int pixel_rep;       // leave at zero
+   int frame_rate;      // screen refresh rate in Hz
+   int interlaced;      // leave at zero
+   int pixel_freq;      // clock frequency (width*height*framerate)
+   int aspect_ratio;    // *
+} hdmi_timings_t;
+
 static enum gfx_ctx_api drm_api           = GFX_CTX_NONE;
+static drmModeModeInfo crt_switch_mode;
+
+static bool gfx_ctx_drm_timings_to_mode(const char *timings_str, drmModeModeInfoPtr modeInfo)
+{
+   int ret;
+   hdmi_timings_t timings;
+
+   if(timings_str != NULL && modeInfo != NULL) {
+      ret = sscanf(timings_str, "%d %d %d %d %d %d %d %d %d %d %d %d %d %d %d %d %d",
+                   &timings.h_active_pixels, &timings.h_sync_polarity, &timings.h_front_porch,
+                   &timings.h_sync_pulse, &timings.h_back_porch,
+                   &timings.v_active_lines, &timings.v_sync_polarity, &timings.v_front_porch,
+                   &timings.v_sync_pulse, &timings.v_back_porch,
+                   &timings.v_sync_offset_a, &timings.v_sync_offset_b, &timings.pixel_rep, &timings.frame_rate,
+                   &timings.interlaced, &timings.pixel_freq, &timings.aspect_ratio);
+      if (ret != 17) {
+         RARCH_ERR("[KMS/EGL]: malformed custom mode: %s\n", timings_str);
+         return false;
+      }
+
+      memset(modeInfo, 0, sizeof(drmModeModeInfo));
+      modeInfo->clock = timings.pixel_freq / 1000;
+      modeInfo->hdisplay = timings.h_active_pixels;
+      modeInfo->hsync_start = modeInfo->hdisplay + timings.h_front_porch;
+      modeInfo->hsync_end = modeInfo->hsync_start + timings.h_sync_pulse;
+      modeInfo->htotal = modeInfo->hsync_end + timings.h_back_porch;
+      modeInfo->hskew = 0;
+      modeInfo->vdisplay = timings.v_active_lines;
+      modeInfo->vsync_start = modeInfo->vdisplay + (timings.v_front_porch * (timings.interlaced ? 2 : 1));
+      modeInfo->vsync_end = modeInfo->vsync_start + (timings.v_sync_pulse * (timings.interlaced ? 2 : 1));
+      modeInfo->vtotal = modeInfo->vsync_end + (timings.v_back_porch * (timings.interlaced ? 2 : 1));
+      modeInfo->vscan = 0; // TODO: ??
```

Comment:
I'm not sure if this value should be set/used, it doesn't seems to be the case in the "[rpi drm driver mode conversion](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/drivers/gpu/drm/drm_modes.c#L587)"